README.md

```
@@ -91,39 +91,75 @@ See [CLAUDE.md - Database Implementation Status](CLAUDE.md#database-implementati
 
+**Keeping requirements.txt in sync with pyproject.toml**:
+
+The `requirements.txt` file is generated from `pyproject.toml` and should be regenerated when dependencies change:
+
+```bash
+# Regenerate requirements.txt from pyproject.toml
+uv pip compile pyproject.toml -o requirements.txt
+```
+
+> **Note**: Consider adding a CI/pre-commit check to ensure `requirements.txt` stays synchronized with `pyproject.toml`. Add this to `.pre-commit-config.yaml` to automatically fail the build if they're out of sync.
+
```

markdownlint-cli2 (0.20.0)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)